File: platforms/react-native/sample/src/accessibility/__tests__/accessibilityIdentifiers.test.ts
import {AccessibilityIdentifiers} from '../accessibilityIdentifiers';

describe('AccessibilityIdentifiers', () => {
  it('pins selector values consumed by Maestro flows', () => {
    expect(AccessibilityIdentifiers.appReady).toBe('checkout-kit-sample-ready');
    expect(AccessibilityIdentifiers.tabs.catalog).toBe('catalog-tab');
    expect(AccessibilityIdentifiers.tabs.cart).toBe('cart-tab');
    expect(AccessibilityIdentifiers.cart.checkoutReady).toBe(
      'cart-checkout-ready',
    );
    expect(AccessibilityIdentifiers.cart.checkoutButton).toBe(
      'checkout-button',
    );
    expect(AccessibilityIdentifiers.cart.emptyMessage).toBe(
      'cart-empty-message',
    );
  });

  it('pins generated Settings selector values used outside TypeScript', () => {
    expect(AccessibilityIdentifiers.settings.section('authentication')).toBe(
      'settings-section-authentication',
    );
    expect(
      AccessibilityIdentifiers.settings.buyerIdentityOption('customerAccount'),
    ).toBe('settings-buyer-identity-option-customer-account');
    expect(AccessibilityIdentifiers.settings.themeOption('storefront')).toBe(
      'settings-theme-option-storefront',
    );
    expect(
      AccessibilityIdentifiers.settings.applePayStyleOption('whiteOutline'),
    ).toBe('settings-apple-pay-style-option-white-outline');
    expect(AccessibilityIdentifiers.settings.checkoutPreloadingSwitch).toBe(
      'settings-checkout-preloading-switch',
    );
  });
});

File: platforms/android/samples/CheckoutKitAndroidDemo/app/src/test/java/com/shopify/checkoutkit/androiddemo/accessibility/AccessibilityIdentifiersTest.kt
package com.shopify.checkoutkit.androiddemo.accessibility

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class AccessibilityIdentifiersTest {
    @Test
    fun `app ready marker matches the maestro flows`() {
        assertThat(AccessibilityIdentifiers.APP_READY).isEqualTo("checkout-kit-sample-ready")
    }

    @Test
    fun `cart markers match the maestro flows`() {
        assertThat(AccessibilityIdentifiers.Cart.CHECKOUT_READY).isEqualTo("cart-checkout-ready")
        assertThat(AccessibilityIdentifiers.Cart.CHECKOUT_BUTTON).isEqualTo("checkout-button")
        assertThat(AccessibilityIdentifiers.Cart.EMPTY_MESSAGE).isEqualTo("cart-empty-message")
    }

    @Test
    fun `tab markers match the maestro flows`() {
        assertThat(AccessibilityIdentifiers.Tabs.CART).isEqualTo("cart-tab")
        assertThat(AccessibilityIdentifiers.Tabs.SETTINGS).isEqualTo("settings-tab")
    }

    @Test
    fun `settings markers match the maestro flows`() {
        assertThat(AccessibilityIdentifiers.Settings.CHECKOUT_PRELOADING_TOGGLE)
            .isEqualTo("checkout-preloading-toggle")
    }
}
